### PR TITLE
fix: modify single scan feature

### PR DESCRIPTION
**Pull request by m-rudyk**
# Pull Request

## Description

1. add loading diff before analyzing files for single scan;
2. add manual database init before using;

## Type of change

Please delete options that are not relevant.

- [X] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] Code cleanup/refactoring
- [x] Documentation update
- [x] This change requires a documentation update
- [ ] CI system update
- [X] Test Coverage update

# Checklist:

- [x] My code follows the style guidelines of this project
- [x] My code meets the required code coverage for lines (90% and above)
- [x] My code meets the required code coverage for branches (80% and above)
- [x] I have performed a self-review of my own code
- [x] I have commented my code, particularly in hard-to-understand areas
- [x] I have made corresponding changes to the documentation
- [x] My changes generate no new warnings
- [x] I have added tests that prove my fix is effective or that my feature works
- [x] New and existing unit tests pass locally with my changes

### CODE REVIEW

File: src/main/java/com/lpvs/service/LPVSDetectService.java [hunk 1]
```diff
@@ -152,12 +138,11 @@ public void runOneScan() {
...
                     log.info(report);
                 }
+                log.info("Single scan completed.");
             } catch (Exception ex) {
                 log.error("\n\n\n Single scan finished with errors \n\n\n");
-                log.error("Can't triger single scan " + ex);
+                log.error("Can't trigger single scan: " + ex.getMessage());
             }
-            // exiting application
-            eventPublisher.publishEvent(new ExitCodeEvent(new Object(), 0));
```
Comment (o-kopysov) [inline src/main/java/com/lpvs/service/LPVSDetectService.java:160]: Why do you skip exiting from the app?

File: src/main/resources/application.properties [hunk 2]
```diff
@@ -14,12 +14,13 @@ license_conflict=db
 
 # Logger configuration
 logging.pattern.console=%d{dd-MM-yyyy HH:mm:ss.SSS} %magenta([%thread]) %highlight(%-5level) %logger.%M - %msg%n
+spring.profiles.active=singlescan
 
 # GitHub settings
 # Corresponding env. variable LPVS_GITHUB_LOGIN
-github.login=
+github.login=m-rudyk
```
Comment (o-kopysov) [inline src/main/resources/application.properties:21]: Why do you place your credentials here?

File: src/test/java/com/lpvs/service/LPVSDetectServiceTest.java [hunk 3]
```diff
@@ -139,8 +135,8 @@ void testRunOneScan_Default() throws NoSuchFieldException, IllegalAccessExceptio
                     spy(new LPVSDetectService("scanoss", null, scanossDetectService, null, null));
 
             setPrivateField(lpvsDetectService, "trigger", "fake-trigger-value");
-            setPrivateField(lpvsDetectService, "eventPublisher", mockEventPublisher);
-            doNothing().when(mockEventPublisher).publishEvent(any());
+            // setPrivateField(lpvsDetectService, "eventPublisher", mockEventPublisher);
+            // doNothing().when(mockEventPublisher).publishEvent(any());
```
Comment (o-kopysov) [inline src/test/java/com/lpvs/service/LPVSDetectServiceTest.java:139]: Why do you place the commented code here?

**Review by o-kopysov — CHANGES_REQUESTED** [review]
I do not see any updates compared to the diff I provided.

File: src/main/java/com/lpvs/service/LPVSDetectService.java (hunk 1 above)
Comment (m-rudyk) [reply src/main/java/com/lpvs/service/LPVSDetectService.java:160]: When we do sys call with System.exit(0) we have JVM exit. 
First - it causes critical flaw in application (we cannot run this through security checker)
Second - it closes JVM during runtime both on build instal and testt phases

Comment (o-kopysov) [reply src/main/java/com/lpvs/service/LPVSDetectService.java:160]: OK. Will be added in a separate PR.

File: src/main/resources/application.properties [hunk 4]
```diff
@@ -14,6 +14,7 @@ license_conflict=db
 
 # Logger configuration
 logging.pattern.console=%d{dd-MM-yyyy HH:mm:ss.SSS} %magenta([%thread]) %highlight(%-5level) %logger.%M - %msg%n
+spring.profiles.active=singlescan
```
Comment (o-kopysov) [inline src/main/resources/application.properties:17]: This line is not required here.
Also, please modify the next property in `application-singlescan.properties`:
spring.datasource.url=jdbc:h2:mem:singlescan;DATABASE_TO_UPPER=false;MODE=MySQL;NON_KEYWORDS=USER;DB_CLOSE_ON_EXIT=FALSE

Comment (m-rudyk) [reply src/main/resources/application.properties]: done

**Review by o-kopysov — APPROVED** [review]
LGTM

**Review by o-konoval — APPROVED** [review]
Great job for fix the issue.  Approved.

**Pull request merged**